app/src/main/java/com/example/mytrainingpal/screens/TrainingsPreview.kt

```diff
@@ -0,0 +1,172 @@
+package com.example.mytrainingpal.screens
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.FitnessCenter
+import androidx.compose.runtime.*
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.navigation.NavController
+import com.example.mytrainingpal.components.*
+import com.example.mytrainingpal.model.entities.Exercise
+import com.example.mytrainingpal.model.entities.Muscle
+import com.example.mytrainingpal.model.view_models.ExerciseMuscleMapViewModel
+import com.example.mytrainingpal.model.view_models.MusclePainEntryMapViewModel
+import com.example.mytrainingpal.model.view_models.MusclePainEntryViewModel
+import com.example.mytrainingpal.states.RememberAddingSoreMusclesToList
+import com.example.mytrainingpal.states.RememberFetchMusclePainEntryWithMuscles
+import com.example.mytrainingpal.states.RememberTodaysMusclePainEntryState
+import com.example.mytrainingpal.util.ExerciseDetails
+import com.example.mytrainingpal.util.IntHolder
+
+
+@Composable
+fun TrainingsPreviewScreen(
+    navController: NavController,
+    duration: IntHolder,
+    exercises: MutableList<Pair<Exercise, ExerciseDetails>>,
+    musclePainEntryViewModel: MusclePainEntryViewModel,
+    musclePainEntryMapViewModel: MusclePainEntryMapViewModel,
+    exerciseMuscleMapViewModel: ExerciseMuscleMapViewModel,
+) {
+    // get the current muscle pains
+    val todaysMusclePainEntry = RememberTodaysMusclePainEntryState(musclePainEntryViewModel)
+    RememberFetchMusclePainEntryWithMuscles(
+        todaysMusclePainEntry = todaysMusclePainEntry,
+        musclePainEntryMapViewModel = musclePainEntryMapViewModel
+    )
+    val soreMuscles: SnapshotStateList<Pair<Muscle, Long>> = remember { mutableStateListOf() }
+    RememberAddingSoreMusclesToList(musclePainEntryMapViewModel, soreMuscles)
+
+    val allExercises by exerciseMuscleMapViewModel.allExercisesWithMuscles.observeAsState(listOf())
+
+    if (allExercises.isNotEmpty()) {
```

Comment:
Please wrap this whole if in a remember statement and let it depend on `allExercises` and `soreMuscles`.
Look at https://github.com/MichaelBuessemeyer/MyTrainingsPal/blob/5512f0740ea8362e7d391ac8d27fbf5554a2faa5/app/src/main/java/com/example/mytrainingpal/states/RememberAddingSoreMusclesToList.kt#L21 for a reference. Remember works as a caching wrapper. The recalculation is always done, when a argument in the ()-brackets changes.

Using remember here will also fix, that the selected exercises change unwantedly when a recomposition of the composable is done (for any reason).

Reply:
nice, thanks!